module/src/main/scala/com/gu/googleauth/auth.scala

```diff
@@ -108,6 +108,10 @@ object GoogleAuth {
           googleResponse(response) { json =>
               val token = Token.fromJson(json)
               val jwt = token.jwt
+              config.domain foreach { domain =>
+                if (!jwt.claims.email.split("@").lastOption.contains(domain))
```

Comment:
Contains is an exact match - it's an option and you're checking the Option contains that value